portfolio/src/main/java/com/google/sps/PuzzleDao.java

```diff
@@ -85,4 +92,35 @@ public void delete(Long puzzleId) {
       //deletes the entity whose id match "puzzleID"
       datastore.delete(keyFactory.newKey(puzzleId));
     }
+
+    public ListPuzzles listPuzzles(String cursorUrl, int PAGE_SIZE) {
+      //queries the puzzles
+      EntityQuery.Builder queryBuilder = Query.newEntityQueryBuilder()
+        .setKind("Puzzle")
+        .setLimit(PAGE_SIZE);
+
+      if(cursorUrl != null){
+        Cursor pageCursor = Cursor.fromUrlSafe(cursorUrl);
+        queryBuilder.setStartCursor(pageCursor);
+      }        
+      
+      //creates a list of puzzles using the querie results
+      ArrayList<Puzzle> puzzles = new ArrayList<Puzzle>();
```

Comment:
The type paramter can be omitted here:
```suggestion
      ArrayList<Puzzle> puzzles = new ArrayList<>();
```